.concourse/scripts/t-cargo-build.sh

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env bash
+
+set -e
+set -x
+set -o pipefail
+
+GIT_DIR=$1
+CACHE_DIR=$2
+ARTIFACT_DIR=$3
+SEMVER_DIR=$4
+
+SEMVER=$(cat $SEMVER_DIR/version)
+GIT_REF=""
+if [[ -f ${GIT_DIR}/.git/short_ref ]]; then
+    GIT_REF=$(cat ${GIT_DIR}/.git/short_ref)
+elif [[ -f "${GIT_DIR}/.git/resource/head_sha" ]]; then
+    GIT_REF=$(cat ${GIT_DIR}/.git/resource/head_sha)
+else
+    echo "no reference for the polymesh version found"
+    exit 1
+fi
+
+# Prime the build cache directory
+mkdir -p "${CACHE_DIR}"
+if [ ! -f ".git/resource/changed_files" ] || grep -v '^.concourse\|^Dockerfile\|^scripts/cli' ".git/resource/changed_files" || [ ! -f "target/release/polymesh" ]; then
+    touch ${CACHE_DIR}/.new_binary
+else
+    rm -f ${CACHE_DIR}/.new_binary
+fi
+
+pushd .
+cd $GIT_DIR
+# Compile if any of the following conditions is met:
+#  - This is a branch merge, not a PR
+#  - This is a PR where the source minus exceptions changed
+#  - The polymesh binary is missing
+if [ ! -f ".git/resource/changed_files" ] || grep -v '^.concourse\|^Dockerfile\|^scripts/cli' ".git/resource/changed_files" || [ ! -f "target/release/polymesh" ]; then
+    rm -f target/release/polymesh
+    cargo +$TOOLCHAIN build --release || cargo build -j 1 --release
```

Comment:
`cargo build -j 1` is not required. I added this to circleci because the build process was running out of ram on multi-threaded builds and it wasn't possible to get more ram on our current plan. If we run into the same issue on our CI, we can make use of memory-optimized instances.